ros/src/tl_detector/tl_detector.py

```diff
@@ -87,37 +92,49 @@ def image_cb(self, msg):
             self.upcoming_red_light_pub.publish(Int32(self.last_wp))
         self.state_count += 1
 
+    def distance(self, p1, p2):
+        """
+        Distance between x and y coordinates modified version of waypoint_lodaer.py.
+        """
+        x, y = p1.x - p2.x, p1.y - p2.y
+        return math.sqrt(x*x + y*y)
```

Comment:
Note: if your just comparing numbers like you are in this code, since you only care about magnitudes and don't need the values for other calculations - an optimization that can be done is to not perform the square root; just return the sum of the squares. Taking the square root is an expensive process, especially for an interpreted language like python (not as bad as some languages, though).

Reply:
For eliminating possible future bus I think it is 2 ways:
- leave at as is (since name need to match purpose)
- rename method into something descriptive and remove sqrt
I think leaving this as is is also good solution since we are working on some educational prototype :)

Reply:
Thanks for reviewing my code. Yes, I think for our initiate state for this project is to make it working perfectly before optimize all the code. The code you have review might be need to be optimize and I'm thinking to optimize it after finished all the task. If you have any idea to optimize the code, feel free to optimize it when our autonomous car working perfectly!